### ISSUE TITLE
Backslash escaped incorrectly in octal character codes in PDF string literals

### ISSUE THREAD

**Issue by shreevatsa** [J-F-Liu/lopdf#147]
If I create a `lopdf::Object::String` like, say,

```
let my_vec: Vec<u8> = r"\115".as_bytes().to_vec();
assert_eq!(my_vec, vec![92, 49, 49, 53]); // BACKSLASH, DIGIT ONE, DIGIT ONE, DIGIT FIVE
lopdf::Object::String(my_vec, lopdf::StringFormat::Literal)
```

then when `lopdf` writes out the PDF file, it writes `\\115` into the PDF file (having 5 bytes instead of 4), because it escapes the backslash. But the backslash shouldn't be escaped when it is followed by any of the standard escape sequences documented in [Table 3 in section "7.3.4.2 Literal Strings" in the PDF 1.7 spec](https://www.adobe.com/content/dam/acom/en/devnet/pdf/pdfs/PDF32000_2008.pdf#page=23):

![image](https://user-images.githubusercontent.com/767530/127911656-8916b26a-8d6d-4098-bbd2-adb7a23d2a30.png)

In this case I'd like to use the last option in the above table, and really write `\115` into the file (well, in this particular case I could just write `M` as octal `115` = decimal `77` = hex `4D` = ASCII `M`, but that doesn't work for arbitrary octal escapes). But lopdf writes `\\115` instead.

-----

I think the issue comes from the following code, where it looks like *all* backslashes are escaped:

https://github.com/J-F-Liu/lopdf/blob/fec4729625312da00243b58db2169168fe4ed65d/src/writer.rs#L154-L184

**Comment by shreevatsa** [comment]
A workaround is to use `StringFormat::Hexadecimal` with the right data, e.g. instead of the four bytes in `\115`, use the byte 77 (or `0o115` or `0x4D` or whatever) directly.

**Comment by J-F-Liu** [comment]
This is not a bug, input r"\115" and encode as `\\115`, so to get '\115' when decoded.
You may write `lopdf::Object::String(vec![0o115], lopdf::StringFormat::Literal)` too.

**Comment by shreevatsa** [comment]
To add some context, I was trying to get the literal four characters `\115` into the PDF file (as part of `/ActualText`). As an example, when I want the ActualText to be "क्" which is U+0915 followed by U+094D, my first attempt was to write something that would put the six bytes `0xfe, 0xff, 0x09, 0x15, 0x09, 0x4D` into the PDF file as `/ActualText (\376\377\011\025\011\115)` and this seems hard to do, which is why I reported this as a bug... 

But thanks for pointing it out, it doesn't *need* to be in that form and 

```
lopdf::Object::String(vec![0o376, 0o377, 0o011, 0o025, 0o011, 0o115], lopdf::StringFormat::Literal)
```

works just fine! In the meantime I started using something like

```
lopdf::Object::String(vec![0xfe, 0xff, 0x09, 0x15, 0x09, 0x4D], lopdf::StringFormat::Hexadecimal)
```

(yes I know it's the same vector, just written in hexadecimal) which is also equivalent.

So maybe no one really *needs* precise control on getting octal escape sequences into PDF string literals, so feel free to close this. 

Thanks again for this very useful library (I basically ended up learning a bit of Rust just to use it),
